### PR TITLE
Fixes: Issues with report endpoints and runs filtering.

### PR DESCRIPTION
**Pull request by ihor-palii**
Copying/Creating a flow step with error using the mouse right-click
https://app.asana.com/0/1159876841343611/1200159103971794/f

Issue: Filtering  flow results by before or after<date>on Runs tab
https://app.asana.com/0/0/1200176098496218/f

Issue: Unable to display more columns after filtering flow results on Runs tab
https://app.asana.com/0/0/1200175217079125/f

Issue: Sentry error when searching flow results by non-existent contact fields
https://app.asana.com/0/0/1200175217079128/f

Issue: Contacts Report: Get Body Fields section on API Explorer should be updated to show the other available fields to use to query
https://app.asana.com/0/0/1200175217079119/f

Issue: Contacts Report: Sending invalid request body does not return 4XX status and 'detail' message
https://app.asana.com/0/0/1200175217079118/f

Issue: Contacts Report: total_unique_contacts returns count for partial group name search
https://app.asana.com/0/0/1200175217079117/f

Issue: Contacts Report: 504 Gateway Timeout error when querying a lot of contacts using API Explorer
https://app.asana.com/0/0/1200151125756187/f

### CODE REVIEW

File: temba/api/v2/views.py [hunk 1]
```diff
@@ -4410,6 +4414,114 @@ def request__get_separated_names_and_uuids(self, field_name):
...
+        if any(values_fo_filter_by.values()):
+            self.applied_filters[field_name] = dict(item for item in values_fo_filter_by.items() if item[1])
+        return queryset
+
+    def get_contacts(self, count_only=False):
+        org = self.request.user.get_org()
+        contacts = Contact.objects.filter(org=org, status=Contact.STATUS_ACTIVE, is_active=True).distinct()
+
+        # filter by channel
+        contacts = self.name_uuid_filtering(contacts, "channel", "urns__channel__uuid", "urns__channel__name")
```
Comment (ihor-palii) [inline temba/api/v2/views.py:4436]: Here we have a joining of two extra tables, that probably can be indexed by elasticsearch.
 `contacts_contact` -> `contacts_contacturn` -> `channels_channel`

File: temba/api/v2/views.py [hunk 2]
```diff
@@ -4410,6 +4414,114 @@ def request__get_separated_names_and_uuids(self, field_name):
...
+
+        # filter by flow
+        flow_uuid = self.request.GET.get("flow", self.request.data.get("flow"))
+        if flow_uuid:
+            if not is_uuid_valid(flow_uuid):
+                raise SearchException(_("The `flow` parameter is not valid."))
+            contacts = contacts.filter(runs__flow__uuid=flow_uuid)
```
Comment (ihor-palii) [inline temba/api/v2/views.py:4443]: Also here we have a joining of two extra tables to filter by, that probably can be indexed by elasticsearch.
`contacts_contact` -> `flows_flowrun` -> `flows_flow`

**Review by teehamaral — APPROVED** [review]

File: temba/api/v2/views.py (hunk 2 above)
Comment (teehamaral) [reply temba/api/v2/views.py:4443]: Yep, can we use the celery for it, having a cronjob to index it overnight daily? I think that could work and would be less time than make the changes on the RP-Indexer for it, I guess.

File: temba/api/v2/views.py (hunk 1 above)
Comment (teehamaral) [reply temba/api/v2/views.py:4436]: Yep, can we use the celery for it, having a cronjob to index it overnight daily? I think that could work and would be less time than make the changes on the RP-Indexer for it, I guess.

**Pull request merged**